src/hotspot/share/runtime/continuation.hpp

```diff
@@ -47,9 +47,26 @@ void continuations_init();
 class javaVFrame;
 class JavaThread;
 
+// should match Continuation.toPreemptStatus() in Continuation.java
+enum freeze_result {
+  freeze_ok = 0,
+  freeze_ok_bottom = 1,
+  freeze_pinned_cs = 2,
+  freeze_pinned_native = 3,
+  freeze_pinned_monitor = 4,
+  freeze_exception = 5,
+  freeze_not_mounted = 6,
+  freeze_unsupported = 7
+};
+
 class Continuation : AllStatic {
 public:
 
+  enum preempt_kind {
+    freeze_on_monitorenter = 1,
```

Comment:
There was one value that meant to be for the regular freeze from java. But it was not used so I removed it.

Reply:
Fair enough, but I would prefer if you start at zero. Just so people like me don't start scratching their head trying to figure out the cosmic reason for why it doesn't start at zero.

Reply:
Yes, I missed to include it in the previous changes. I actually removed the assignment altogether since there is no need to rely on particular values (although it will start at zero by default).